ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -449,9 +475,85 @@ public boolean terminateAllQueries() {
     return true;
   }
 
+  public void removeTemporaryQuery(QueryMetadata queryMetadata) {
+    this.allLiveQueries.remove(queryMetadata);
+  }
+
   public DDLCommandResult executeDdlStatement(final DDLStatement statement,
                                               final Map<String, Object> streamsProperties) {
     return queryEngine.handleDdlStatement(statement, streamsProperties);
   }
 
+  private class KsqlEngineMetrics implements Closeable {
+    private final String metricGroupName;
+    private final Sensor numActiveQueries;
+    private final Sensor messagesIn;
+    private final Sensor messagesOut;
+
+    KsqlEngineMetrics(String metricGroupPrefix) {
+      Metrics metrics = MetricCollectors.getMetrics();
+
+      this.metricGroupName = metricGroupPrefix + "-query-stats";
+      this.numActiveQueries = metrics.sensor(metricGroupName + "-active-queries");
+      numActiveQueries.add(
+          metrics.metricName("num-active-queries", this.metricGroupName,
+                             "The current number of active queries running in this engine"),
+          new MeasurableStat() {
+            @Override
+            public double measure(MetricConfig metricConfig, long l) {
+              return allLiveQueries.size();
```

Comment:
my only concern with this is that if we want to capture more stats, i.e. min, max, percentile then you will need to revert to the KafkaMetric construction rather than anon-class the interface as you have here.  I guess that can be dealt with later

Reply:
We can update in the future if we need the 'min', 'max', etc. 

But given that this is a point in time total value which will be collected periodically, I am not sure that a 'min', 'max' etc. will be too useful.